src/sentry/models/dynamicsampling.py

```diff
@@ -0,0 +1,258 @@
+        app_label = "sentry"
+        db_table = "sentry_customdynamicsamplingruleproject"
+        unique_together = (("custom_dynamic_sampling_rule", "project"),)
+
+
+@region_silo_only_model
+class CustomDynamicSamplingRule(Model):
+    """
+    This represents a custom dynamic sampling rule that is created by the user based
+    on a query (a.k.a. investigation rule).
+
+    """
+
+    __relocation_scope__ = RelocationScope.Organization
+
+    date_added = models.DateTimeField(default=timezone.now)
+    organization = FlexibleForeignKey("sentry.Organization", on_delete=models.CASCADE)
+    projects = models.ManyToManyField(
+        "sentry.Project",
+        related_name="custom_dynamic_sampling_rules",
+        through=CustomDynamicSamplingRuleProject,
+    )
+    is_active = models.BooleanField(default=True)
+    is_org_level = models.BooleanField(default=False)
+    rule_id = models.IntegerField(default=0)
+    condition = models.TextField()
+    sample_rate = models.FloatField(default=0.0)
+    start_date = models.DateTimeField(default=timezone.now)
+    end_date = models.DateTimeField()
+    num_samples = models.IntegerField()
+    condition_hash = models.CharField(max_length=40)
+
+    @property
+    def external_rule_id(self) -> int:
+        """
+        Returns the external rule id
+
+        For external users, i.e. Relay, we need to shift the ids since the slot we
+        have allocated starts at the offset specified in RESERVED_IDS.
+        """
+        return self.rule_id + CUSTOM_RULE_START
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_customdynamicsamplingrule"
+        indexes = [
+            # get active rules for an organization
+            models.Index(fields=["organization"], name="org_idx", condition=Q(is_active=True)),
+            # get expired rules (that are still marked as active)
+            models.Index(fields=["end_date"], name="end_date_idx", condition=Q(is_active=True)),
+            # find active rules for a condition
+            models.Index(
+                fields=["condition_hash"], name="condition_hash_idx", condition=Q(is_active=True)
+            ),
+        ]
+
+    @staticmethod
+    def get_rule_for_org(
+        condition: Any, organization_id: int
+    ) -> Optional["CustomDynamicSamplingRule"]:
+        """
+        Returns an active rule for the given condition and organization if it exists otherwise None
+
+        Note: There should not be more than one active rule for a given condition and organization
+        This function doesn't verify this condition, it just returns the first one.
+        """
+        condition_hash = get_condition_hash(condition)
+        rules = CustomDynamicSamplingRule.objects.filter(
+            organization_id=organization_id,
+            condition_hash=condition_hash,
+            is_active=True,
+            end_date__gt=timezone.now(),
+        )
+        for rule in rules:
+            return rule
+        return None
```

Comment:
```suggestion
        return rules[0] if rules else None
```

Reply:
Not sure how the QuerySet responds to this ( i.e. `if rules:`), if it was a list it would make sense, 
it may work with QuerySet as well but I didn't try it.

Reply:
Checked, it works, changed